source/AI.cpp

```diff
@@ -2923,6 +2923,25 @@ Point AI::TargetAim(const Ship &ship, const Body &target)
 
 
 
+bool AI::TargetInRange(const Ship &ship, const Hardpoint &hardpoint)
+{
+	shared_ptr<const Body> shipTarget = ship.GetTargetShip();
+	shared_ptr<const Body> targetAsteroid = ship.GetTargetAsteroid();
+	shared_ptr<const Body> target = shipTarget ? shipTarget : targetAsteroid;
+
+	if(!target)
+		return false;
```

Comment:
If the player has no target and wants to fire, then they shouldn't be stopped from firing. This can be the case if the player is trying to mine but doesn't have an asteroid scanner, and so can't actually target asteroids.